src/App.jsx

```diff
@@ -1,13 +1,37 @@
-import { useState } from 'react'
+import { useState, useEffect } from 'react'
+import { Routes, Route } from 'react-router-dom';
+import Loading from './Page-Components/Loading'
 import './css/App.css'
+import Header from './Components/Header'
+import Home from './Page-Components/Home'
+import getAllArticles from './api';
+import AllArticles from './Page-Components/AllArticles';
+
 
 function App() {
+  const [articles, setArticles] = useState([])
+  const[isLoading, setIsLoading] = useState(true)
+
+  useEffect(() => {
+     getAllArticles()
+     .then((articles) => {setArticles(articles)})
+     setIsLoading(false)
+  }, []);
 
-  return (
-    <>
-     <h1>NC News</h1>
-    </>
-  )
-}
+    if(isLoading){
```

Comment:
Great work rendering your app conditionally depending on whether isLoading is set to true. This really enhances the user experience